kornia/geometry/quaternion.py

```diff
@@ -74,7 +74,7 @@ def __repr__(self) -> str:
         return f"real: {self.real} \nvec: {self.vec}"
 
     def __getitem__(self, idx):
-        return self.data[idx]
+        return Quaternion(self.data[idx][None, :])
```

Comment:
Not very clear why the None?

Reply:
since Quaternion accepts batched tensors. Perhaps change it to reshape(1, -1) ?